src/load-config.js

```diff
@@ -54,7 +54,10 @@ export default {
   hostname: process.env.ACTUAL_HOSTNAME || config.hostname,
   serverFiles: process.env.ACTUAL_SERVER_FILES || config.serverFiles,
   userFiles: process.env.ACTUAL_USER_FILES || config.userFiles,
-  webRoot: process.env.ACTUAL_WEB_ROOT || config.webRoot,
+  webRoot: path.resolve(
+    projectRoot,
+    process.env.ACTUAL_WEB_ROOT || config.webRoot
+  ),
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏the more I think about this, the more I dislike it from a security POV.

If an attacker gains access to this config file or the env variables: he basically can traverse the entire filesystem. I imagine setting it to `../../../path/to/budget.data?ignore=` and then opening up the server would spit out the budget file. 

https://www.acunetix.com/websitesecurity/directory-traversal/

Reply:
If the attacker gains access to the config file then they presumably already have access to the filesystem, right?

And the environment variables are evaluated once at startup, so changes to them at runtime would not have any effect. If you were able to set them before server startup, you could already change the `ACTUAL_SERVER_FILES` environment variable to an empty directory, then re-bootstrap the app and download all of the files with the normal UI.

Reply:
Not necessarily. There might be another system running on the same instance. And the attacker might be able to upload files to it, but not read from it.

Anyways, that's a lot of _ifs_. However, we do need to consider them and try and keep the attack vector for actual-server as little as possible. Directory traversal attack is a pretty common security issue. So if we can treat **all** the input as potentially malicious - we can protect ourselves better.

In this case I really don't like the wrapping of this input in `path.resolve`. Is there an alternative we could consider that would solve your problem? Perhaps symlinks?